File: src/test/java/io/gluth/pagespace/domain/PageTest.java
package io.gluth.pagespace.domain;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class PageTest {

    @Test
    void storesIdAndTitle() {
        Page page = new Page("id1", "Title One");
        assertEquals("id1", page.id());
        assertEquals("Title One", page.title());
    }

    @Test
    void equalById() {
        Page a = new Page("same", "Alpha");
        Page b = new Page("same", "Beta");
        assertEquals(a, b);
        assertEquals(a.hashCode(), b.hashCode());
    }

    @Test
    void notEqualForDifferentIds() {
        Page a = new Page("id1", "Alpha");
        Page b = new Page("id2", "Alpha");
        assertNotEquals(a, b);
    }

    @Test
    void nullIdThrows() {
        assertThrows(NullPointerException.class, () -> new Page(null, "Title"));
    }

    @Test
    void nullTitleThrows() {
        assertThrows(NullPointerException.class, () -> new Page("id", null));
    }
}

File: src/test/kotlin/io/gluth/pagespace/domain/PageTest.kt
package io.gluth.pagespace.domain

import org.junit.jupiter.api.Test
import kotlin.test.assertEquals
import kotlin.test.assertNotEquals

class PageTest {

    @Test
    fun storesIdAndTitle() {
        val page = Page("id1", "Title One")
        assertEquals("id1", page.id)
        assertEquals("Title One", page.title)
    }

    @Test
    fun equalById() {
        val a = Page("same", "Alpha")
        val b = Page("same", "Beta")
        assertEquals(a, b)
        assertEquals(a.hashCode(), b.hashCode())
    }

    @Test
    fun notEqualForDifferentIds() {
        val a = Page("id1", "Alpha")
        val b = Page("id2", "Alpha")
        assertNotEquals(a, b)
    }
}
